### PR TITLE
Adding line-height setting & granular font size.

### PR DESCRIPTION
**Pull request by jakearchibald**
Added line-height setting as per https://github.com/dawnlabs/carbon/issues/406.

While I was there, I also changed font-size so it increments by 0.5, since the steps were pretty big.

### CODE REVIEW

File: lib/routing.js [hunk 1]
```diff
@@ -32,6 +32,7 @@ const mappings = [
   { field: 'ln:lineNumbers', type: 'bool' },
   { field: 'fm:fontFamily' },
   { field: 'fs:fontSize' },
+  { field: 'lh:lineHeight' },
```
Comment (jakearchibald) [inline lib/routing.js]: I'm cargo-culting here. Is this right?

File: lib/constants.js [hunk 2]
```diff
@@ -481,6 +481,7 @@ export const DEFAULT_SETTINGS = {
   language: DEFAULT_LANGUAGE,
   fontFamily: 'Hack',
   fontSize: '14px',
+  lineHeight: '133%',
```
Comment (jakearchibald) [inline lib/constants.js]: The app currently uses `4 / 3` as the line height. This is the closest value that's an integer.

File: components/Slider.js [hunk 3]
```diff
@@ -13,6 +13,7 @@ export default class extends React.Component {
   render() {
     const minValue = this.props.minValue || 0
     const maxValue = this.props.maxValue || 100
+    const step = 'step' in this.props ? this.props.step : 1;
```
Comment (jakearchibald) [inline components/Slider.js]: 0 is a valid step value, so a simple truthy check isn't good enough.

File: components/Settings.js [hunk 4]
```diff
@@ -69,6 +70,14 @@ class Settings extends React.Component {
...
           />
           <Collapse label="Advanced">
+            <Slider
+              label="Line height"
+              value={this.props.lineHeight}
+              minValue={90}
+              maxValue={250}
+              usePercentage={true}
+              onChange={this.props.onChange.bind(null, 'lineHeight')}
+            />
```
Comment (jakearchibald) [inline components/Settings.js]: Happy to move this, but I put it to the top so it's closest to the font-size control while still being in the 'advanced' settings.

**Comment by jakedex** [comment]
This is great! Thanks, @jakearchibald 👍

**Pull request merged**